jwst/datamodels/fits_support.py

```diff
@@ -489,7 +495,7 @@ def callback(schema, path, combiner, ctx, recurse):
         if schema.get('type') == 'array':
             has_fits_hdu = _schema_has_fits_hdu(schema)
             if has_fits_hdu:
-                for i in range(len(hdulist)):
+                for i in range(max_extver):
```

Comment:
So help me understand what was and is now happening. Say you have a file that had 5 tuples of (sci,err,dq), for a total of 15 hdu's (16 counting the primary). In this case `len(hdulist)=16`, while `max_extver=5`, and hence you're reducing the number of searches by a factor of 3?

Reply:
And we have a winner! Yep, that is exactly it. The [original issue](https://jira.stsci.edu/browse/JP-1343) has a full explanation, including suggestions on where to go from here.

Reply:
Amazing that such a simple 2-3 line change can get us a factor 7x reduction in processing time. Awesome!

Reply:
Makes me worry about how many other similar instances are lurking all over the depths of the datamodels code ...